### PR TITLE
fix: hide error box when there is 0 issues :bug:

### LINKED ISSUE #5632: Remove message and icon when the infrastructure has no errors and warnings
### Who would benefit from this feature?

End users

### What is this feature about?

- Delete the message and the icon when the infrastructure does not contain errors and warnings 
image

### Why is this feature valuable?

delete useless informations 

### Anything else people should know?

_No response_

### PR DESCRIPTION
**Pull request by tykayn**
hide messages showing there is no anomaly when there is 0 issues in the editor

close #5632

### CODE REVIEW

**Review by woshilapin — APPROVED** [review]
Looks good and if Codecov is happy, I'm too!

**Comment by Khoyo** [comment]
Please follow our commit style guide: https://osrd.fr/en/docs/guides/contribute/code/#git-commit-style

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .DS_Store
 *.backup
 __pycache__
+front/tests/unit/coverage
```
Comment (clarani) [inline .gitignore]: missing new line

File: codecov.yml [hunk 2]
```diff
@@ -30,3 +30,6 @@ flags:
     paths:
       - front/
     carryforward: true
+
+ignore:
+  - "front/src/tests"
```
Comment (clarani) [inline codecov.yml]: Missing new line

File: front/tests/005-operational-studies.spec.ts [hunk 3]
```diff
@@ -9,6 +9,14 @@ import { ProjectPage } from './pages/project-page-model';
 import { StudyPage } from './pages/study-page-model';
 
 test.describe('Testing if all mandatory elements simulation configuration are loaded in operationnal studies app', () => {
+  test('Testing there is no error', async ({ page }) => {
```
Comment (clarani) [inline front/tests/005-operational-studies.spec.ts]: Thanks for this test :pray:
However you put your test in the operational studies test file but this is not related to operational studies, can you move it to a new file, editor.spec.ts ? Thanks :)

File: front/tests/pages/home-page-model.ts [hunk 4]
```diff
@@ -45,6 +47,7 @@ export class PlaywrightHomePage {
     this.getBody = page.locator('body');
     this.translation = home;
     this.getViteOverlay = page.locator('vite-plugin-checker-error-overlay');
+    this.getErrorBox = page.locator('.error-box');
```
Comment (clarani) [inline front/tests/pages/home-page-model.ts]: Same here, can you move this code to a new file, editor-page ?

**Review by clarani — COMMENTED** [review]
Thanks for your PR :pray:

File: .gitignore (hunk 1 above)
Comment (flomonster) [reply .gitignore]: You probably want to configure you're editor. For vscode:

```json
{
  "files.insertFinalNewline": true,
}
```

Comment (tykayn) [reply .gitignore]: yup, we can also use the [.editorconfig standard](https://editorconfig.org) to autoconfigure any IDE 

```
# Unix-style newlines with a newline ending every file
[*]
end_of_line = lf
insert_final_newline = true
```

Comment (tykayn) [reply .gitignore]: updated with a new line in the end

File: codecov.yml (hunk 2 above)
Comment (tykayn) [reply codecov.yml]: updated with a new line in the end of codecov too

File: front/tests/005-operational-studies.spec.ts (hunk 3 above)
Comment (tykayn) [reply front/tests/005-operational-studies.spec.ts]: i did put it in a dedicated file

File: front/tests/pages/home-page-model.ts (hunk 4 above)
Comment (tykayn) [reply front/tests/pages/home-page-model.ts]: yup, done, created a file for the editor page

**Review by shenriotpro — CHANGES_REQUESTED** [review]
Any chance you can rebase to remove irrelevant commits from this PR?

**Comment by tykayn** [comment]
> Any chance you can rebase to remove irrelevant commits from this PR?

yup @shenriotpro i made a rebase and force pushed

**Comment by shenriotpro** [comment]
> > Any chance you can rebase to remove irrelevant commits from this PR?
> 
> yup @shenriotpro i made a rebase and force pushed

Great, thanks

**Review by shenriotpro — APPROVED** [review]

File: front/.gitignore [hunk 5]
```diff
@@ -7,6 +7,7 @@
 
 # testing
 /coverage
+front/tests/unit/coverage
```
Comment (Khoyo) [inline front/.gitignore]: Out of scope for this pr

File: front/.yarnrc.yml [hunk 6]
```diff
@@ -0,0 +1 @@
+nodeLinker: node-modules
```
Comment (Khoyo) [inline front/.yarnrc.yml]: Out of scope

**Review by Khoyo — COMMENTED** [review]
The yarn.lock update seems out of scope too.

All of these should be done in separate PRs, or at least separate commits.

File: front/tests/005-operational-studies.spec.ts
Comment (flomonster) [inline front/tests/005-operational-studies.spec.ts:1]: This seems to be a reformat of the whole file. Can you put this in a separate commit?

File: front/.gitignore (hunk 5 above)
Comment (tykayn) [reply front/.gitignore]: ok removed

File: front/.yarnrc.yml (hunk 6 above)
Comment (tykayn) [reply front/.yarnrc.yml]: ok removed

File: front/tests/005-operational-studies.spec.ts
Comment (tykayn) [reply front/tests/005-operational-studies.spec.ts:1]: yup it was just a reformat. i am tempted to add an editor config to have it consistent, but that would be the subject of a chore ticket. i removed this formatting from the PR.

**Review by Yohh — COMMENTED** [review]
yop, I'm glad to see a new test file for our infra editor :muscle: , can you add `/editor/` in the `grep` key of the playwright config in `playwright.config.ts` ? if you don't add it the test can't be launch

**Review by Yohh — COMMENTED** [review]
Also, can you rename the files related to the tests by specifying that it is the infra editor? (and not the rolling stock editor)

**Review by Yohh — APPROVED** [review]
nice! :muscle:

**Review by nicolaswurtz — CHANGES_REQUESTED** [review]
Hi! Indentation is not correct, it must be 2 spaces.

File: front/playwright.config.ts [hunk 7]
```diff
@@ -62,7 +62,7 @@ const config: PlaywrightTestConfig = {
 
   /* select tags to run specific tests */
   // TODO: remove grep when every tests are refactored
-  grep: [/home/, /project/, /study/, /scenario/],
+  grep: [/editor/, /home/, /project/, /scenario/, /study/],
```
Comment (nicolaswurtz) [inline front/playwright.config.ts]: please respect order : home / project / study / scenario / editor  thx

**Review by nicolaswurtz — CHANGES_REQUESTED** [review]
please split in 2 commits:
- one for bug correction
- one for adding test

File: front/playwright.config.ts (hunk 7 above)
Comment (tykayn) [reply front/playwright.config.ts]: ok updated the order and split in two commits

Comment (woshilapin) [reply front/playwright.config.ts]: Any reason for this order? I find the alphabetical order more logical to respect. If there is a specific need for another order, maybe a comment would help?

Comment (Yohh) [reply front/playwright.config.ts]: in a future pr the grep will be associated with the keyword `enable` only

Comment (nicolaswurtz) [reply front/playwright.config.ts]: My bad, I'v not seen that it was ordered in alphabetical order (sorry @tykayn!). No passion for me, so do whatyou think that'll better (original order was logical order of views hierarchy, not so obvious!)

**Comment by tykayn** [comment]
After a request from @nicolaswurtz i am splitting the PR in two, one for the feature, and one to test the feature:
https://github.com/osrd-project/osrd/pull/6414

**Review by nicolaswurtz — APPROVED** [review]
